packages/core/src/node/filesystem-locking.ts

```diff
@@ -0,0 +1,89 @@
+// *****************************************************************************
+// Copyright (C) 2023 Ericsson and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { Mutex } from 'async-mutex';
+import { injectable, interfaces } from 'inversify';
+import type { URI } from '../common';
+import { FileUri } from './file-uri';
+import path = require('path');
+
+export const FileSystemLocking = Symbol('FileSystemLocking') as symbol & interfaces.Abstract<FileSystemLocking>;
+/**
+ * Use this backend service to help prevent race access to files on disk.
+ */
+export interface FileSystemLocking {
+    /**
+     * Get exclusive access to a file for reading and/or writing.
+     * @param lockPath The path to request exclusive access to.
+     * @param transaction The job to do while having exclusive access.
+     * @param thisArg `this` argument used when calling `transaction`.
+     */
+    lockPath<T>(lockPath: string | URI, transaction: (lockPath: string) => T | Promise<T>, thisArg?: unknown): Promise<T>;
+    /**
+     * Check if someone already has exclusive access to a file.
+     * @param lockPath The path to test if access is restricted.
+     */
+    isPathLocked(lockPath: string | URI): boolean;
```

Comment:
It doesn't seem that this is every referred to, and it seems that the intended API is to lock the path and then perform whatever operation (reading or writing), that needs to be done?

Reply:
This is a left over, will remove.